app/models/namespace.rb

```diff
@@ -111,6 +130,88 @@ def ransackable_attributes(_auth_object = nil)
     def ransackable_associations(_auth_object = nil)
       %w[]
     end
+
+    def subtract_from_metadata_summary_count(namespaces, metadata_summary, by_one)
+      return if metadata_summary.empty?
+
+      update_metadata_summary_counts(namespaces, metadata_summary, by_one: by_one, addition: false)
+    end
+
+    def add_to_metadata_summary_count(namespaces, metadata_summary, by_one)
+      return if metadata_summary.empty?
+
+      update_metadata_summary_counts(namespaces, metadata_summary, by_one: by_one, addition: true)
+    end
+
+    private
+
+    def update_metadata_summary_counts(namespaces, metadata_summary, by_one: false, addition: true) # rubocop:disable Metrics/AbcSize,Metrics/MethodLength
+      metadata_summary_update_sql = build_metadata_summary_update_sql(metadata_summary,
+                                                                      addition ? Arel::Nodes::Addition : Arel::Nodes::Subtraction,
+                                                                      by_one:)
+
+      entry_table = Arel::Table.new(
+        Arel::Nodes::NamedFunction.new(
+          'jsonb_each',
+          [Namespace.arel_table[:metadata_summary].concat(metadata_summary_update_sql)]
+        ),
+        as: 'entry'
+      )
+
+      updated_metadata_summary = entry_table.project(
```

Comment:
This one too.

Reply:
Refactored and added comments in [f7f508a](https://github.com/phac-nml/irida-next/pull/1343/commits/f7f508a1b981044a40806c3d934fd06a9565400d), will leave larger refactor of repeated Arel calls to a followup PR.